tests/tracking/test_rest_tracking.py

```diff
@@ -1236,6 +1236,28 @@ def test_create_model_version_with_non_local_source(mlflow_client, monkeypatch):
     assert response.status_code == 400
     assert "If supplying a source as an http, https," in response.json()["message"]
 
+    response = requests.post(
+        f"{mlflow_client.tracking_uri}/api/2.0/mlflow/model-versions/create",
+        json={
+            "name": name,
+            "source": "mlflow-artifacts://host:9000/models/..%2f..%2fartifacts",
+            "run_id": run.info.run_id,
+        },
+    )
+    assert response.status_code == 400
+    assert "If supplying a source as an http, https," in response.json()["message"]
```

Comment:
Can we also add test cases for multiple slashes in the URL and many dots (e.g. 4 dots)?

Reply:
Multiple slashes will be converted to single slash, and it should work fine (there's a test case already). For many dots, the current logic is still passing because it's not violating any rules, but the url "xxxx/..../" is not a correct path. Should we just add limitation to converting more than two dots into two dots? Or not allowing them at all?

Reply:
I think multiple dots is a valid filename, so we should allow it - just want to make sure that there isn't an LFI with that. I think it's fine